diff -Nru oldsrc/emu/render.cpp src/emu/render.cpp
--- oldsrc/emu/render.cpp	2020-08-06 22:26:32.000000000 +1000
+++ src/emu/render.cpp	2020-08-10 18:02:46.486027381 +1000
@@ -1356,7 +1356,8 @@
 		}
 
 	// if we are not in the running stage, draw an outer box
-	else
+	// DISABLE WHITE BORDER
+	else if ( !m_manager.machine().options().skip_gameinfo() )
 	{
 		render_primitive *prim = list.alloc(render_primitive::QUAD);
 		set_render_bounds_xy(prim->bounds, 0.0f, 0.0f, (float)m_width, (float)m_height);
diff -Nru oldsrc/frontend/mame/ui/mainmenu.cpp src/frontend/mame/ui/mainmenu.cpp
--- oldsrc/frontend/mame/ui/mainmenu.cpp	2020-08-06 22:26:32.000000000 +1000
+++ src/frontend/mame/ui/mainmenu.cpp	2020-08-10 18:02:46.490027425 +1000
@@ -129,6 +129,9 @@
 
 	item_append(menu_item_type::SEPARATOR);
 
+	// DISABLE SELECT NEW MACHINE & FAVORITES
+	if (!machine().options().skip_gameinfo())
+	{
 	if (!mame_machine_manager::instance()->favorite().is_favorite(machine()))
 		item_append(_("Add To Favorites"), "", 0, (void *)ADD_FAVORITE);
 	else
@@ -143,6 +146,7 @@
 //  item_append(_("Quit from Machine"), nullptr, 0, (void *)QUIT_GAME);
 
 	item_append(_("Select New Machine"), "", 0, (void *)SELECT_GAME);
+	}
 }
 
 menu_main::~menu_main()
diff -Nru oldsrc/frontend/mame/ui/ui.cpp src/frontend/mame/ui/ui.cpp
--- oldsrc/frontend/mame/ui/ui.cpp	2020-08-06 22:26:33.000000000 +1000
+++ src/frontend/mame/ui/ui.cpp	2020-08-10 18:03:34.162562672 +1000
@@ -311,6 +311,8 @@
 	bool show_warnings = true, show_mandatory_fileman = true;
 	bool video_none = strcmp(downcast<osd_options &>(machine().options()).video(), OSDOPTVAL_NONE) == 0;
 
+	// FORCE INTERACTIVE WARNING MESSAGES (HARD RESET)
+	first_time = show_gameinfo;
 	// disable everything if we are using -str for 300 or fewer seconds, or if we're the empty driver,
 	// or if we are debugging, or if there's no mame window to send inputs to
 	if (!first_time || (str > 0 && str < 60*5) || &machine().system() == &GAME_NAME(___empty) || (machine().debug_flags & DEBUG_FLAG_ENABLED) != 0 || video_none)
@@ -801,6 +803,10 @@
 
 uint32_t mame_ui_manager::handler_messagebox(render_container &container)
 {
+	// DISABLE INITIALIZING, LOADING & DECRYPTING MESSAGES
+	if (machine().options().skip_gameinfo())
+		return 0;
+
 	draw_text_box(container, messagebox_text.c_str(), ui::text_layout::LEFT, 0.5f, 0.5f, messagebox_backcolor);
 	return 0;
 }
