diff -Nru oldsrc/frontend/mame/ui/mainmenu.cpp src/frontend/mame/ui/mainmenu.cpp
--- oldsrc/frontend/mame/ui/mainmenu.cpp	2021-06-01 15:30:38.000000000 +1100
+++ src/frontend/mame/ui/mainmenu.cpp	2021-06-01 20:26:36.499791000 +1100
@@ -130,6 +130,9 @@
 
 	item_append(menu_item_type::SEPARATOR);
 
+	// DISABLE SELECT NEW MACHINE & FAVORITES
+	if (!machine().options().skip_gameinfo())
+	{
 	if (!mame_machine_manager::instance()->favorite().is_favorite(machine()))
 		item_append(_("Add To Favorites"), 0, (void *)ADD_FAVORITE);
 	else
@@ -144,6 +147,7 @@
 //  item_append(_("Quit from Machine"), 0, (void *)QUIT_GAME);
 
 	item_append(_("Select New Machine"), 0, (void *)SELECT_GAME);
+	}
 }
 
 menu_main::~menu_main()
diff -Nru oldsrc/frontend/mame/ui/ui.cpp src/frontend/mame/ui/ui.cpp
--- oldsrc/frontend/mame/ui/ui.cpp	2021-06-01 15:30:38.000000000 +1100
+++ src/frontend/mame/ui/ui.cpp	2021-06-01 22:56:04.551160587 +1100
@@ -208,6 +208,8 @@
 			ui_callback_type::GENERAL,
 			[this] (render_container &container) -> uint32_t
 			{
+				// DISABLE INITIALIZING, LOADING & DECRYPTING MESSAGES
+				if (!machine().options().skip_gameinfo())
 				draw_text_box(container, messagebox_text, ui::text_layout::LEFT, 0.5f, 0.5f, colors().background_color());
 				return 0;
 			});
@@ -409,6 +411,8 @@
 	bool show_warnings = true, show_mandatory_fileman = true;
 	bool video_none = strcmp(downcast<osd_options &>(machine().options()).video(), OSDOPTVAL_NONE) == 0;
 
+	// FORCE INTERACTIVE WARNING MESSAGES (HARD RESET)
+	first_time = show_gameinfo;
 	// disable everything if we are using -str for 300 or fewer seconds, or if we're the empty driver,
 	// or if we are debugging, or if there's no mame window to send inputs to
 	if (!first_time || (str > 0 && str < 60*5) || &machine().system() == &GAME_NAME(___empty) || (machine().debug_flags & DEBUG_FLAG_ENABLED) != 0 || video_none)
